### PR TITLE
Initial RM CI Pipelines

### PR DESCRIPTION
**Pull request by AdamHawtin**
### Motivation and Context
Pipeline for deploying the RM services to a CI environment in kubernetes. There are two functions of the pipeline, the first is to run through the kubernetes configuration files for the RM services, deploying them based on master in the census-rm-kubernetes repo. The second is to check for new builds of each RM service image latest tag in the container registry, then runs a `kubectl patch` with a timestamp to trigger kubernetes to pull the newly built image. 

The pipeline in running in the census concourse and has successfully deployed and updated the RM services.

As a stop gap measure the jobs currently have a hardcoded wait after running to give the services time to start up. This is limited in that it won't fail the concourse job if the deployment fails to start up, but it does prevent the deployments from running on top of each other which has caused issues. The aim is to replace this with dynamic waits which actually check the service availability which can then fail the job accordingly if the deployment does not start up in a given time.

### What has Changed
- New pipeline to deploy RM to a CI/dev environment
- New example secrets
- Updated README
- Updated gitignore

### Links
https://trello.com/c/pnfYl0em/435-concourse-13
https://github.com/ONSdigital/census-rm-terraform/pull/5

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[Concourse (13)](https://trello.com/c/pnfYl0em/435-concourse-13)

File: README.md [hunk 1]
```diff
@@ -1,2 +1,27 @@
-# census-rm-deploy
-Response Management deployment pipelines and documentation 
+# Census RM Deploy
+Response management Concourse pipelines and documentation
+
+## Pipelines
+### CI Kubernetes Pipeline
+Specified in `pipelines/ci-kubernetes-pipeline.yml`, this pipeline has two functions.
+
+The first pulls from the `census-rm-kubernetes` repo triggered by commits to master and applies the latest microservice and handler configs to the kubernetes cluster specified by the pipeline configuration.
```
Comment (davidmort) [inline README.md]: could make "census-rm-kubernetes" a link to the repo

File: README.md [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+
+The second checks for new builds for the `latest` tag for the RM container registry images. On being triggered by a new build, it runs through the applying the config from master, then runs `kubectl patch` for the newly built images deployment giving them a timestamp and image digest label. This causes kubernetes to re-pull the latest image and bring up new pods while we are using `imagePullPolicy: Always`.
+
+#### How to fly
+Create a secrets YAML file containing the required configuration:
+
+| Variable | Format | Description |
+| --- | --- | ---: |
+| kubernetes-server | https://\<host>:\<port> | The URL of the kubernetes server endpoint |
```
Comment (davidmort) [inline README.md]: Might be worth mentioning "master" in the description, just to be specific.....

File: README.md [hunk 3]
```diff
@@ -1,2 +1,27 @@
...
+| kubernetes-namespace | String | The kubernetes namespace for concourse to deploy to |
+| kubernetes-token | JWT String | A JWT access token. Must be for an account with the required roles in kubernetes |
```
Comment (davidmort) [inline README.md]: might be worth being a bit more specific and mention "service account" not just "account" in the description

**Review by davidmort — CHANGES_REQUESTED** [review]

**Review by davidmort — APPROVED** [review]

**Comment by kierenbeckett** [comment]
`kubectl rollout status deployment <deployment>`

**Review by davidmort — APPROVED** [review]

**Review by liamtoozer — APPROVED** [review]

**Pull request merged**